packages/gatsby-plugin-gatsby-cloud/src/indicator.js

```diff
@@ -0,0 +1,342 @@
+        fill="white"
+      />
+    </svg>
+  </>
+)
+
+const FailedIcon = () => (
+  <>
+    <svg
+      width="15"
+      height="15"
+      viewBox="0 0 20 20"
+      fill="none"
+      xmlns="http://www.w3.org/2000/svg"
+      id="FailedIcon"
+    >
+      <circle cx="10" cy="10" r="10" fill="#FA2915" />
+      <path
+        d="M14.6663 6.2735L13.7263 5.3335L9.99967 9.06016L6.27301 5.3335L5.33301 6.2735L9.05967 10.0002L5.33301 13.7268L6.27301 14.6668L9.99967 10.9402L13.7263 14.6668L14.6663 13.7268L10.9397 10.0002L14.6663 6.2735Z"
+        fill="white"
+      />
+    </svg>
+  </>
+)
+
+export default class Indicator extends React.Component {
+  state = {
+    attributes: {},
+    icon: ``,
+  }
+
+  componentDidMount() {
+    const prettyUrlRegex = /^preview-/
+    const host = window.location.hostname
+    let buildId
+
+    try {
+      setInterval(
+        async () => {
+          // currentBuild is the most recent build that is not QUEUED
+          // latestBuild is the most recent build that finished running (ONLY status ERROR or SUCCESS)
+          const { currentBuild, latestBuild } = await getBuildInfo()
+
+          if (!buildId) {
+            if (prettyUrlRegex.test(host)) {
+              buildId = latestBuild?.id
+            } else {
+              const buildIdMatch = host.match(/build-(.*?(?=\.))/)
+              buildId = buildIdMatch && buildIdMatch[1]
+            }
+          }
+
+          if (currentBuild?.buildStatus === `BUILDING`) {
+            this.setState(prevState =>
+              Object.assign({}, prevState, {
+                attributes: indicatorSetBuilding(),
+                icon: <Spinner />,
+              })
+            )
+          } else if (currentBuild?.buildStatus === `ERROR`) {
+            this.setState(prevState =>
+              Object.assign({}, prevState, {
+                attributes: indicatorSetFailed(),
+                icon: <FailedIcon />,
+              })
+            )
+          } else if (buildId === currentBuild?.id) {
+            this.setState(prevState =>
+              Object.assign({}, prevState, {
+                attributes: indicatorSetUpToDate(),
+                icon: ``,
+              })
+            )
+          } else if (
+            buildId !== latestBuild?.id &&
+            latestBuild?.buildStatus === `SUCCESS`
+          ) {
+            this.setState(prevState =>
+              Object.assign({}, prevState, {
+                attributes: indicatorSetSuccess(latestBuild?.id),
+                icon: <SuccessIcon />,
+              })
+            )
+          }
+        },
+        process.env.NODE_ENV === `test` ? 10 : POLLING_INTERVAL
+      )
+    } catch (e) {
+      console.log(e)
+    }
+  }
+
+  render() {
+    if (Object.keys(this.state.attributes).length === 0) {
+      return <>{this.props.children}</>
```

Comment:
Tiny suggestion!

```suggestion
      return this.props.children
```

Reply:
Thanks for taking a look through all of this! I haven't had a chance to go through and clean up some of the code yet. In terms of sharing helpers, we would have to publish a public package for these style helpers I believe. I don't think that is done anywhere though. In this case though, I didn't want to spend too long thinking about it because I know the design will change drastically pretty soon and the majority of the style/UI code will need to be redone.